USDA_data.py

```diff
@@ -1,38 +1,31 @@
-import os
+import matplotlib.pyplot as plt
 import pandas as pd
+import os
+ 
+# Read input data into dataframe df
+df = pd.read_csv('USDA_data.csv', low_memory=False)
 
-# Data will be outputed to ./results directory
-os.makedirs('./results', exist_ok=True)
-
-# Read input data as DataFrame df
-df = pd.read_excel('USDA_data.xlsx')
-
-# Clean data by removing baselining and duplicates
-df = df[df['Usage'] != 'Baselining'].drop_duplicates()
-
-# Ignore application that skews visualizations
-df = df[df['Name'] != 'Adobe Reader and Acrobat Manager']
-
-# Indicate all columns where tag data may be located
-id_columns = [
-    'Asset - Custom Tags - Copy.2', 
-    'Asset - Custom Tags - Copy.3', 
-    'Asset - Custom Tags - Copy.4', 
-    'Asset - Custom Tags - Copy.5'
-    ]
-
-# Save all unique Mission Areas and Agency IDs into Set tags
+# Part 2: Save all unique Mission Areas and Agency IDs into set tags
 tags = set()
-for column in id_columns:
+# Indicate all columns where tag data may be located and save unique tags into a set 'tags'
+IDColumns = ['Asset - Custom Tags - Copy.2', 'Asset - Custom Tags - Copy.3', 'Asset - Custom Tags - Copy.4', 'Asset - Custom Tags - Copy.5']
+for column in IDColumns:
     tags.update(df[column].dropna().unique())
+# Make folders to store figures and data
+os.mkdir("./data")
+os.mkdir("./figures")
 
-# Iterate through tags
+# Part 3: Iterate through tags and create a 'tag_indicator' to signal if the data is within the tag's domain
 for tag in tags:
-    # 'tag_indicator' signals if the data has the given tag
     df['tag_indicator'] = False
-    for column in id_columns:
+    for column in IDColumns:
         df.loc[df[column] == tag, 'tag_indicator'] = True
-    # Create DataFrame grouped by the 'Name' and 'Usage' for the given tag
-    df_usage = df.loc[df['tag_indicator']].groupby(['Name', 'Usage'], as_index=False).size()
-    # Export to '/results/TAG_usage.xlsx' Excel file
-    df_usage.to_excel("./results/" + tag + "_usage.xlsx", index=False)
+    # Create a df_usage dataframe grouped by the 'Name' and 'Usage' of every 'tag' and export as a new 'tag_usage' excel
+    df_usage = df.loc[df['tag_indicator']==True].groupby(['Name', 'Usage']).size().reset_index()
+    # Pivot the dataframe to display the software usage by usage levels
+    df_usage.rename(columns={df_usage.columns[2]: "Values" }, inplace = True)
+    df_usage = df_usage.pivot(index='Name', columns='Usage', values="Values")
+    df_usage.to_excel("./data/" + tag + "_usage.xlsx")
+    # Create bar grahs and save as PNGs
+    fig = df_usage.plot(kind="bar")
+    fig.figure.savefig("./figures/" + tag + "_image.png")
```

Comment:
I'm still getting the error below:

```
RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
  fig = self.plt.figure(figsize=self.figsize)
```

It might be useful to take a look at `matplotlib.pyplot.close`.